uaclient/conftest.py

```diff
@@ -57,15 +57,18 @@ def _warn_about_new_version():
 
 
 @pytest.yield_fixture(scope="session", autouse=True)
-def util_we_are_currently_root():
+def util_we_are_currently_root(request):
     """
     A fixture that mocks util.we_are_currently_root for all tests.
     Default to true as most tests need it to be true.
     """
-    from uaclient.util import we_are_currently_root
+    from uaclient import util
 
-    original = we_are_currently_root
-    with mock.patch("uaclient.util.we_are_currently_root", return_value=True):
+    is_root = getattr(request, "param", True)
+    with mock.patch(
+        "uaclient.util.we_are_currently_root", return_value=is_root
+    ):
+        original = util.we_are_currently_root
```

Comment:
This isn't needed. You can mock.patch('uaclient.util.we_are_currently_root`) to return a different value as normal for a unit test if you need it to return False.

Reply:
I wanted it to return both True and False in a parametrized way,  instead of having the test duplicated with different return values.

Reply:
Yep! Check [this](https://github.com/canonical/ubuntu-pro-client/blob/main/uaclient/tests/test_security.py#L2267-L2274) out for an example of this

Reply:
Thanks for the link.